editor/editor_file_dialog.cpp

```diff
@@ -436,11 +452,23 @@ void EditorFileDialog::_action_pressed() {
 			return;
 		}
 
+		bool one_more_question = false;
+		if (!dir_access->dir_exists(dir_access->get_current_dir())) {
+			if (!can_create_dir) {
+				exterr->set_text("Folder does not exist.");
+				exterr->popup_centered_minsize(Size2(250, 80) * EDSCALE);
+				return;
+			}
+			confirm_save->set_text("Folder does not exist, Create?");
+			confirm_save->popup_centered(Size2(200, 80));
+			one_more_question = true;
+		}
 		if (dir_access->file_exists(f) && !disable_overwrite_warning) {
 			confirm_save->set_text(TTR("File Exists, Overwrite?"));
 			confirm_save->popup_centered(Size2(200, 80));
-		} else {
-
+			one_more_question = true;
+		}
+		if (!one_more_question) {
```

Comment:
IMO `if / else if / else` would be clearer than this `one_more_question` bool :)

Reply:
(Since I doubt you can have both non existing dir and existing file at the same time.)